addons/web/static/src/views/fields/x2many/x2many_field.js

```diff
@@ -191,15 +192,14 @@ export class X2ManyField extends Component {
                 }
             });
 
-        const editable = archInfo.editable || this.props.editable;
         props.activeActions = this.activeActions;
         props.archInfo = { ...archInfo, columns };
         props.cycleOnTab = false;
-        props.editable = !this.props.readonly && editable;
         props.nestedKeyOptionalFieldsData = this.nestedKeyOptionalFieldsData;
         props.onAdd = (params) => {
             params.editable =
-                !this.props.readonly && ("editable" in params ? params.editable : editable);
+                !this.props.readonly &&
+                ("editable" in params ? params.editable : archInfo.editable || this.props.editable);
```

Comment:
still uncertain with this mess

Reply:
You could have keep this as is, I think (the child will call onAdd, and has the information in archInfo)

Reply:
@aab-odoo I don't even understand why the editable value would change on add.

Reply:
in SectionList or something like that: a list with two kinds of records, one of them is editable inline, the other isn't. The support for this is new, you can ask dam about it